File: examples/perl/.harmont/pipeline.ts
import { pipeline, push, type PipelineDefinition } from "harmont";
import { perl } from "harmont/toolchains";

const project = perl({ path: "." });

const pipelines: PipelineDefinition[] = [
  {
    slug: "ci",
    triggers: [push({ branch: "main" })],
    pipeline: pipeline(project.test(), project.lint(), {
      env: { CI: "true" },
      defaultImage: "ubuntu:24.04",
    }),
  },
];

export default pipelines;

File: examples/cpp/.harmont/pipeline.py
"""C++ example pipeline."""
from __future__ import annotations

import harmont as hm


@hm.pipeline(
    "ci",
    env={"CI": "true"},
    default_image="ubuntu:24.04",
    triggers=[hm.push(branch="main")],
)
def ci() -> tuple[hm.Step, ...]:
    project = hm.cmake(path=".", defines={"CMAKE_BUILD_TYPE": "Release", "CMAKE_CXX_STANDARD": "17"})
    return (
        project.test(),
        project.lint(),
        project.fmt(),
    )
